cfme/infrastructure/datastore.py

```diff
@@ -241,21 +267,17 @@ def get_vms(self):
             view.entities.relationships.click_at('VMs')
         else:
             view.entities.relationships.click_at('Managed VMs')
-        # todo: to replace with correct view
-        vms_view = view.browser.create_view(DatastoresView)
-        return [vm.name for vm in vms_view.entities.get_all()]
+        vms_view = view.browser.create_view(DatastoreManagedVMsView)
```

Comment:
@valaparthvi i don't think you need to create this view if we have navigation destination for `MangedVMs`
```
    view = navigate_to(self, "ManagedVMs")
```

Reply:
If I navigate to `ManagedVMs`, it will click on "Managed VMs" and then navigate, but there is a condition where it clicks on "VMs" instead of "Managed VMs", I think navigation would fail at this condition.

Reply:
@valaparthvi and @digitronik let's continue this discussion, I don't think it needs to block this PR. Should the conditional selection of `VMs` or `Managed VMs` done here be in the `step` method for the navigation class, so that it can be used reliably?